pyteal/ast/subroutine.py

```diff
@@ -21,15 +34,13 @@ def __init__(
         self,
         implementation: Callable[..., Expr],
         returnType: TealType,
-        nameStr: str = None,
+        nameStr: Optional[str] = None,
+        routerRegistrable: bool = False,
     ) -> None:
         super().__init__()
         self.id = SubroutineDefinition.nextSubroutineId
         SubroutineDefinition.nextSubroutineId += 1
-
-        self.by_ref_args: Set[str] = set()
```

Comment:
I see this declaration is being removed. I think it's still safe -and preferable- to keep a version of it. In general it's nice to know what all the fields are in a class along with their types as early as possible in the init. If you don't want to initialize the member here you can still do the following:
```python
self.by_ref_args: Set[str]
```

Reply:
yes, I was just having a havoc on the code and was too lazy to annotate them for additional knowledge of type 😅

I will fix that soon.

Reply:
OK I have them initialized on line 69 and 72!

Reply:
Similar comment regarding `expected_arg_types`

Reply:
I guess it's also handled